[DOC] Add how-to guides for local/HPC parallelization (#713)
nipoppy/nipoppy: docs/source/how_to_guides/parallelization/local_parallelization.md

# Parallelizing pipeline runs on non-HPC systems

> This guide shows possible ways to parallelize pipeline runs on computer systems that do not have job schedulers supported by the Nipoppy tools.

Reviewer: ```suggestion
This guide shows possible ways to parallelize pipeline runs on computer systems without job schedulers supported by Nipoppy.
```
